### PR TITLE
NAS-124761 / 24.04 / Do not allow changing an alias if it is being consumed by apps

### PR DESCRIPTION
**Pull request by sonicaj**
This commit adds changes to not allow changing an alias if it's being consumed by apps.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/network.py [hunk 1]
```diff
@@ -1066,6 +1066,22 @@ def __validate_aliases(self, verrors, schema_name, data, ifaces):
...
+            return
+
+        aliases = {
+            k: [entry['address'] for entry in data.get('aliases', [])] for k, data in (('old', old), ('new', new))
+        }
+
+        if node_ip in aliases['old'] and node_ip not in aliases['new']:
+            verrors.add(
+                f'{old["id"]}.aliases',
+                f'{node_ip!r} is being consumed by Applications and before changing it, please use a different node IP'
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/network.py:1081]:
```suggestion
                f'{node_ip!r} is being consumed by Applications, please use a different node IP'
```

**Review by yocalebo — CHANGES_REQUESTED** [review]

**Review by yocalebo — APPROVED** [review]

**Review by yocalebo — CHANGES_REQUESTED** [review]
Hmm, actually looks like unit tests need to be updated.

**Comment by sonicaj** [comment]
@yocalebo, yup - fixed

**Review by yocalebo — APPROVED** [review]

**Comment by sonicaj** [comment]
backport

**Pull request merged**